patterns/behavioural/inner-trait.md

```diff
@@ -0,0 +1,105 @@
+# Inner Trait
+
+## Description
+
+It is possible to define a private trait that implements all the
+methods of a public trait and also includes some private functions. This pattern
+can be used to provide additional functionality to the implementation of a
+public trait while keeping the private methods hidden from the public API.
+
+## Example
+
+This example demonstrate how a public trait `Car` can be implemented and include
+extra private methods using a auxiliary private trait `InnerCar`.
+
```

Comment:
I think we probably want to show a real example in standard library or in popular crates for this.